lib/galaxy/tools/wrappers.py

```diff
@@ -80,14 +80,12 @@ def __init__(self, input, value, other_values={}):
         self._other_values = other_values
 
     def __eq__(self, other):
-        if isinstance(other, string_types):
+        if isinstance(other, string_types) or (isinstance(other, InputValueWrapper) and other.input.type == 'text'):
             return str(self) == other
-        elif isinstance(other, int):
-            return int(self) == other
-        elif isinstance(other, float):
+        elif isinstance(other, (int, float)) or (isinstance(other, InputValueWrapper) and other.input.type in ('float', 'integer')):
```

Comment:
I'd keep int and float separate, just to be sure.

Reply:
I think we need to treat them the same, otherwise the last test I added (5 < 5.1) would fail, or did I miss something?

Reply:
Wouldn't the following work?

```suggestion
        elif isinstance(other, int) or (isinstance(other, InputValueWrapper) and other.input.type == 'integer'):
            return int(self) == other
        elif isinstance(other, float) or (isinstance(other, InputValueWrapper) and other.input.type == 'float'):
```

Reply:
If self wraps a float this would lead to wrong results, e.g 5.1 == 5. It would be ok if we made this 
```
elif self.input.type == 'integer' and  isinstance(other, int) or (isinstance(other, InputValueWrapper) and other.input.type == 'integer'):
```
But then we also need to take care of the case where `self.input.type == 'integer'` and other is float-ish ... .
On the other hand I think casting to float should work fine in most/all cases.

Reply:
Or ... we turn this around and cast self to the correct type. That seems like less guesswork.